sklearn/cluster/tests/test_mean_shift.py

```diff
@@ -39,27 +38,34 @@ def test_estimate_bandwidth():
     assert 0.9 <= bandwidth <= 1.5
 
 
-def test_estimate_bandwidth_1sample():
+def test_estimate_bandwidth_1sample(global_dtype):
     # Test estimate_bandwidth when n_samples=1 and quantile<1, so that
     # n_neighbors is set to 1.
-    bandwidth = estimate_bandwidth(X, n_samples=1, quantile=0.3)
-    assert bandwidth == pytest.approx(0.0, abs=1e-5)
+    bandwidth = estimate_bandwidth(
+        X.astype(global_dtype, copy=False), n_samples=1, quantile=0.3
+    )
+
+    assert bandwidth.dtype == X.dtype
+    assert_allclose(bandwidth, 0.0, atol=1e-5)
 
```

Comment:
For me, it would make more sense to use `pytest.approx` when checking scalar values.
They both have absolute and relative tolerance and `assert_allclose` would let me know that we deal with NumPy arrays (or array-like).